src/pybind/mgr/devicehealth/module.py

```diff
@@ -401,26 +401,29 @@ def show_device_metrics(self, devid, sample):
         res = {}
         ioctx = self.open_connection(create_if_missing=False)
         if ioctx:
-            with rados.ReadOpCtx() as op:
-                omap_iter, ret = ioctx.get_omap_vals(op, "", sample or '', 500)  # fixme
-                assert ret == 0
-                try:
-                    ioctx.operate_read_op(op, devid)
-                    for key, value in list(omap_iter):
-                        if sample and key != sample:
-                            break
-                        try:
-                            v = json.loads(value)
-                        except (ValueError, IndexError):
-                            self.log.debug('unable to parse value for %s: "%s"' %
-                                           (key, value))
-                            pass
-                        res[key] = v
-                except rados.ObjectNotFound:
-                    pass
-                except rados.Error as e:
-                    self.log.exception("RADOS error reading omap: {0}".format(e))
-                    raise
+            try:
```

Comment:
i'd suggest remove the check for `ioctx` at https://github.com/ceph/ceph/pull/25058/files#diff-5ec1b43b7338971207fc4f09440710aaR403 . as `Rados.open_ioctx()` always returns an `Ioctx` instance, which never evaluates to something that is interpreted as false.

Reply:
@tchaikov Thanks.